add update parser and make it available to the code editor

deflox opened the pull request.

m-dot-wigger commented on src/components/editor/assembly.ts:
```diff
@@ -6,14 +6,15 @@ export const AssemblyLanguage = LRLanguage.define({
+        VariableName: t.variableName,
+        RegisterLiteral: t.variableName,
+        IntegerLiteral: t.number,
+        MachInstrIntegerLiteral: t.number,
+        MachInstrSymbolicLiteral: t.variableName,
```
i guess you meant Match, right?

deflox replied:
No this is mean to the short name of "Machine".

m-dot-wigger reviewed (changes requested):
I took the file sumdiff of lab 5 (ArithmOperations) as example:
```
; -- sumdiff.s
; --
; -- CT1 P05 Summe und Differenz
; --
; -- $Id: sumdiff.s 705 2014-09-16 11:44:22Z muln $
; ------------------------------------------------------------------
;Directives
        PRESERVE8
        THUMB

; ------------------------------------------------------------------
; -- Symbolic Literals
; ------------------------------------------------------------------
ADDR_DIP_SWITCH_7_0     EQU     0x60000200
ADDR_DIP_SWITCH_15_8    EQU     0x60000201
ADDR_LED_7_0            EQU     0x60000100
ADDR_LED_15_8           EQU     0x60000101
ADDR_LED_23_16          EQU     0x60000102
ADDR_LED_31_24          EQU     0x60000103

; ------------------------------------------------------------------
; -- myCode
; ------------------------------------------------------------------
        AREA MyCode, CODE, READONLY

main    PROC
        EXPORT main

user_prog
        ; STUDENTS: To be programmed
		
		; read operand A
		LDR		R0, =ADDR_DIP_SWITCH_15_8
		LDRB	R0, [R0]
		LSLS	R0, R0, #24

		; read operand B
		LDR		R1, =ADDR_DIP_SWITCH_7_0
		LDRB	R1, [R1]
		LSLS	R1, R1, #24
		
		; CALC SUM
		ADDS	R2, R0, R1
		;save application program status register ! right after since LSRS also sets flags...
		MRS		R3, APSR
		
		;shift back so only most significant byte is in r4
		LSRS	R4, R2, #24
		
		; show most significant byte of sum result on lower right leds
		LDR		R7, =ADDR_LED_7_0
		STRB	R4, [R7]
		
		; STRB damit nur immer das jeweilig Byte geschrieben wird und nicht der Rest auch noch von höheren LEDs überschrieben wird
		
		;show flags of addition
		LDR		R7, =ADDR_LED_15_8
		LSRS	R3, R3, #24
		STRB	R3, [R7]
		
		
		
		;alles mit B zusatz damit nur bytes geschrieben werden und so nicht die leds mit hohem Index weniger stark leuchten, da in Schleife
		; immmer mal wieder nicht leuchten
		
		; CALC SUB: a - b
		SUBS	R2, R0, R1
		;save application program status register
		MRS		R3, APSR
		
		;shift back so only most significant byte is in r4
		LSRS	R4, R2, #24
		
		; show most significant byte of sub result on upper right leds
		LDR		R7, =ADDR_LED_23_16
		STRB	R4, [R7]
		
		;show flags of subtraction
		LDR		R7, =ADDR_LED_31_24
		LSRS	R3, R3, #24
		STRB	R3, [R7]

        ; END: To be programmed
        B       user_prog
        ALIGN
; ------------------------------------------------------------------
; End of code
; ------------------------------------------------------------------
        ENDP
        END
```

Hightlighting looks good but has some flaws:

- it looks like underline (_) is not considered for variable, label and literal names

- ADDR has red color probably of ADD operation

deflox commented:
@m-dot-wigger I know that the syntax highlighting is not perfect. But I just added the parser as far as I implemented it. Since there is still an open story in the backlog that should handle syntax highlighting this can be fixed in this story: https://virtualctboard.atlassian.net/browse/VCB-29

m-dot-wigger commented on src/components/editor/assembly.grammar:
```diff
@@ -1,50 +1,182 @@
+
+@tokens {
+
+  identifier { $[A-Za-z0-9]+ }
```
just add _ here as well please

deflox replied:
done

m-dot-wigger commented on src/components/editor/assembly.grammar:
```diff
@@ -1,50 +1,182 @@
+  kw<"EORS"> |
+  kw<"ORRS"> |
+  kw<"BICS"> |
+  kw<"MVNS"> |
```
Please add kw<"MRS"> as instruction since if I implement the instruction I would not add it here

deflox replied:
done

m-dot-wigger commented on src/components/editor/assembly.grammar:
```diff
@@ -1,50 +1,182 @@
+
+  "[" "]"
+
+}
```
as in .ts files please let last line be an empty one

m-dot-wigger reviewed (changes requested):
some small changes to improve highlighting and remaining parts can be done in linked userstory

deflox commented:
@m-dot-wigger I made the changes as suggested and merged the current master state in the PR. @clooudch Do you also want to take a look as you are actually the designated reviewer? I would propose @m-dot-wigger is enough already since the changes are not that big.
